File: 493.py
import utils
"""
70 colored balls are placed in an urn, 10 for each of the seven colors.
What is the expected number of distinct colors in 20 randomly picked balls?
Give your answer with nine digits after the decimal point (a.bcdefghij).
"""


def init():
    """
        create all the permutations for the number drawn from each bucket

        [[10, 10], [10, 9, 1]]...
    """
    acc = [10, 10]


def randomDraw():
    """
        how many ways are there to choose a given count from the 7 buckets
        eg, given that we want to choose 2 colors, how many ways are there

        10 choose 10 == 1, 7 choose 2 == 21, so 21 (1 * 21)
    """
    print utils.choose(7, 3)


# 3 == 0.7 == 2.62404092072 ; 4 == 42
randomDraw()

File: 493.go
package main

// recursively add each ball option
func drawBall(balls string, rem string, acc *[]string, targetNum int) {
	// check if we're done (have drawn all of them)
	if len(balls) == targetNum {
		*acc = append(*acc, balls)
		return
	}

	// for every remaining ball, create new balls array and keep going
	for i, ball := range rem {
		drawBall(balls, append(rem[:i], rem[i+1:]...), acc, targetNum)
	}
}

// 70 colored balls are placed in an urn, 10 for each of the seven rainbow colors.
// What is the expected number of distinct colors in 20 randomly picked balls?
// Give your answer with nine digits after the decimal point (a.bcdefghij).

// looks like another DP problem... count up chance of each happening
// func main() {

// 	//sinitialize the balls string, with all of them
// 	in := ""
// 	for i := 0; i < 7; i++ {
// 		for j := 0; j < 10; j++ {
// 			in += strconv.Itoa(j)
// 		}
// 	}

// 	// 4 == 11.6
// }
